build.gradle

```diff
@@ -3,16 +3,16 @@ plugins {
     id 'checkstyle'
     id 'pmd'
     id 'jacoco'
-    id 'com.github.johnrengelman.shadow' version '6.1.0'
-    id 'nebula.lint' version '17.7.0'
-    id "org.owasp.dependencycheck" version "7.2.1"
+    id 'maven-publish'
 }
 
 group 'no.unit.bibs'
 version '1.0-SNAPSHOT'
 
-sourceCompatibility = JavaVersion.VERSION_11  // source-code version and must be <= targetCompatibility
-targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
+java {
+    sourceCompatibility = JavaVersion.VERSION_21 // source-code version and must be <= targetCompatibility
+    targetCompatibility = JavaVersion.VERSION_21 // bytecode target version
```

Comment:
we also need to do this in template.yaml and buildspec.yml, along with manually setting CodeBuild image to 7.0 in CodeBuild (manually)